tests/cnf/ran/diskencryption/tests/tpm2.go

```diff
@@ -0,0 +1,342 @@
+		err = cluster.WaitForClusterRecover(Spoke1APIClient, []string{
+			RANConfig.GeneralConfig.MCONamespace}, tsparams.TimeoutClusterRecovery, time.Second)
+	})
+
+	It("Verifies that disabling Secure Boot prevents"+
+		" Disk decryption (TPM PCR 7)", func() {
+
+		By("checking that Root disk is encrypted with tpm2 with PCR 1 and 7")
+		var luksListOutput string
+		luksListOutput, err = helper.GetClevisLuksListOutput()
+		Expect(err).To(BeNil(), "error getting of clevis luks list command")
+		isRootDiskTPM2PCR1AND7 := parsehelper.LuksListContainsPCR1And7(luksListOutput)
+		Expect(err).To(BeNil(), "error when checking if the root disk is configured with TPM 1 and 7")
+		if !isRootDiskTPM2PCR1AND7 {
+			Skip("Root disk is not encrypted with PCR 1 and 7, skip TPM1 tests")
+		}
+
+		By("checking if Secure Boot is enabled")
+		var isSecureBootEnabled bool
+
+		Eventually(func() error {
+			isSecureBootEnabled, err = BMCClient.IsSecureBootEnabled()
+
+			return err
+		}).WithTimeout(tsparams.TimeoutWaitingOnBMC).
+			WithPolling(tsparams.PoolingIntervalBMC).Should(BeNil())
+
+		if !isSecureBootEnabled {
+			Skip("Secure boot is not enabled test cannot proceed")
+		}
+
+		By("checking that the reserved slot is not present")
+		var isRootDiskReservedSlotPresent bool
+		luksListOutput, err = helper.GetClevisLuksListOutput()
+		Expect(err).To(BeNil(), "error getting of clevis luks list command")
+		isRootDiskReservedSlotPresent = parsehelper.LuksListContainsReservedSlot(luksListOutput)
+		Expect(err).To(BeNil(), "error checking if the output of clevis luks list contains a reserved slot")
+		Expect(isRootDiskReservedSlotPresent).To(BeFalse(), "there should be no reserved slot present at this point")
+
+		By("disabling Secure Boot")
+		Eventually(BMCClient.SecureBootDisable).WithTimeout(tsparams.TimeoutWaitingOnBMC).
+			WithPolling(tsparams.PoolingIntervalBMC).Should(BeNil())
+
+		By("restarting node gracefully")
+		err = cluster.SoftRebootSNO()
+		Expect(err).To(BeNil(), "error rebooting node")
+
+		By("waiting for Disk decryption Failure log to appear")
+		matches := []stdinmatcher.Matcher{
+			{
+				Regex: regexp.MustCompile("TPM failed"),
+				Times: 10,
+			},
+			{
+				Regex: regexp.MustCompile("pcr-rebind-boot"),
+				Times: 1,
+			},
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), tsparams.TimeoutWaitRegex)
+		defer cancel()
+		matchIndex, err := stdinmatcher.WaitForRegex(ctx, BMCClient, matches)
+
+		Expect(err).To(BeNil(), "WaitForRegex should not fail")
+		Expect(matchIndex).To(Equal(0), "WaitForRegex should match TPM failed (0)")
+
+		By("enabling SecureBoot")
+		Eventually(BMCClient.SecureBootEnable).WithTimeout(tsparams.TimeoutWaitingOnBMC).
+			WithPolling(tsparams.PoolingIntervalBMC).
+			Should(Or(BeNil(), MatchError("secure boot is already enabled")))
+
+		By("powercycling node")
+		Eventually(BMCClient.SystemPowerCycle).WithTimeout(tsparams.TimeoutWaitingOnBMC).
+			WithPolling(tsparams.PoolingIntervalBMC).Should(BeNil())
```

Comment:
see above about assertion comments and `HaveOccurred()`

Reply:
fixed